test/functional/feature_onlynet.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test -onlynet configuration option.
+"""
+
+from test_framework.basic_server import (
+    BasicServer,
+    tor_control,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+
+class OnlynetTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.extra_args = [
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1'],
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-onion=127.0.0.1:9050'],
+            ['-onlynet=ipv4', '-torcontrol=127.0.0.1:19051', '-listenonion=1', '-proxy=127.0.0.1:9050'],
+        ]
+        self.num_nodes = len(self.extra_args)
+        self.tor_control_server = BasicServer(bind=('127.0.0.1', 19051),
+                                              response_generator=tor_control)
+        # see https://github.com/bitcoin/bitcoin/pull/22651#discussion_r686656841
+        self.setup_clean_chain = True
+
+    def onion_is_reachable(self, node):
+        for net in self.nodes[node].getnetworkinfo()['networks']:
+            if net['name'] == 'onion':
+                return net['reachable']
+        return False
+
+    def run_test(self):
+        # Node 0 would fail the check without https://github.com/bitcoin/bitcoin/pull/22651.
+        # Nodes 1 and 2 pass with or without https://github.com/bitcoin/bitcoin/pull/22651,
+        # they are here just to ensure behavior does not change unintended.
```

Comment:
nits:

- I think it's more robust to describe why than to refer to a GitHub PR or depend on GitHub in general

- nit s/unintended/unintentionally/

Reply:
I agree, removed the github reference.

I think "uninternationally" is even better, changed.